### PR TITLE
doc: v12.0.0 release notes

### PR DESCRIPTION
**Pull request by theanalyst**
Just the draft release notes upto current master, I'll correct the unknown tags once we decide on the final sha of the release, major TODO is the feature announcements and upgrade path for anyone wanting to try.

### CODE REVIEW

File: doc/release-notes.rst [hunk 1]
```diff
@@ -1,6 +1,417 @@
...
  Release Notes
 ===============
+v12.0.0 Luminous (dev)
+======================
+
+This is the first development checkpoint release of Luminous series, the next
+long time release. We're off to a good start to release Luminous in the spring
+of '17.
+
+* UNKNOWN: Add openstack requirements to smoke suite (`pr#12913 <http://github.com/ceph/ceph/pull/12913>`_, Zack Cerza)
```
Comment (liewegas) [inline doc/release-notes.rst]: qa

**Comment by theanalyst** [comment]
@jdurgin @liewegas @yehudasa @dillaman @jcsp any significant announcements we need to make for the release

File: doc/release-notes.rst [hunk 2]
```diff
@@ -1,6 +1,417 @@
...
+* UNKNOWN: Move code from .h into .cc (`pr#12737 <http://github.com/ceph/ceph/pull/12737>`_, Amir Vadai)
```
Comment (liewegas) [inline doc/release-notes.rst]: common:

Comment (theanalyst) [inline doc/release-notes.rst]: I'll get these together and sort in the end

File: doc/release-notes.rst [hunk 3]
```diff
@@ -1,6 +1,417 @@
...
+* UNKNOWN: NVMEDevice.cc: cleanup the logic in data_buf_next_sge (`pr#13056 <http://github.com/ceph/ceph/pull/13056>`_, optimistyzy)
+* UNKNOWN: NVMEDevice: optimize sector_size usage (`pr#12780 <http://github.com/ceph/ceph/pull/12780>`_, optimistyzy)
```
Comment (liewegas) [inline doc/release-notes.rst]: os/bluestore

File: doc/release-notes.rst [hunk 4]
```diff
@@ -1,6 +1,417 @@
...
+* UNKNOWN: async: Fixed compilation error when enable -DWITH_DPDK (`pr#12660 <http://github.com/ceph/ceph/pull/12660>`_, Pan Liu)
+* UNKNOWN: async: fixed coredump when enable dpdk. (`pr#12854 <http://github.com/ceph/ceph/pull/12854>`_, Pan Liu)
```
Comment (liewegas) [inline doc/release-notes.rst]: msg/async:

File: doc/release-notes.rst [hunk 5]
```diff
@@ -1,6 +1,417 @@
...
+* UNKNOWN: async: fixed the error "Cause: Cannot create lock on '/var/run/.rte_c… (`pr#12860 <http://github.com/ceph/ceph/pull/12860>`_, Pan Liu)
```
Comment (liewegas) [inline doc/release-notes.rst]: msg/async

**Comment by theanalyst** [comment]
Scheduled 1/99 suite for rados, rbd,cephfs & rgw. this was luminous atop of `fc2df15`

- _running_ rados http://pulpito.ceph.com/abhi-2017-02-07_09:15:52-rados-wip-luminous-2---basic-smithi/
- _running_ rbd http://pulpito.ceph.com/abhi-2017-02-07_09:21:33-rbd-wip-luminous-2---basic-smithi/
- _running_ fs http://pulpito.ceph.com/abhi-2017-02-07_09:25:06-fs-wip-luminous-2---basic-smithi/
- _running_ krbd http://pulpito.ceph.com/abhi-2017-02-07_09:27:51-krbd-wip-luminous-2---basic-smithi/
- _running_ rgw http://pulpito.ceph.com/abhi-2017-02-07_09:33:57-rgw-wip-luminous-2---basic-smithi/

**Comment by jcsp** [comment]
@theanalyst I just merged https://github.com/ceph/ceph/pull/13259 if you want to pull it into the release notes, no bother if it's awkward to do so

**Comment by theanalyst** [comment]
@jcsp np, I'll pull it in

**Comment by idryomov** [comment]
> FAILED krbd http://pulpito.ceph.com/abhi-2017-02-07_09:27:51-krbd-wip-luminous-2---basic-smithi/
ignored

@theanalyst Just a note for the future -- k{rbd,cephfs,nfs) test runs should be scheduled with `teuthology-suite ... -k testing`.

**Comment by theanalyst** [comment]
@idryomov thanks, will do

**Comment by theanalyst** [comment]
@jcsp seeing some issues with a cephfs teuth run. Is this something already encountered earlier

**Comment by jcsp** [comment]
@theanalyst 

Valgrind: client (Leak_StillReachable)
1 jobs: ['795353']
-> this is a (probably harmless) leak in the lttng setup code, possible related to the recent libcommon changes, I've asked Kefu to look into it but this isn't a release blocker.

Failure: Test failure: test_files_throttle (tasks.cephfs.test_strays.TestStrays)
1 jobs: ['795355']
-> this is not completely understood yet but is not a user-visible bug, so don't block the release.

**Comment by theanalyst** [comment]
@jcsp thanks for letting me know, going ahead with the release then

**Pull request merged**